### ISSUE TITLE
Issue while transfering Native Matic token (Status Pending - This txn hash was found in our secondary node and should be picked up by our indexer in a short while)

### ISSUE THREAD

**Issue by savannahar68** [maticnetwork/maticjs-plasma#5]
I'm running [transfer_matic.js](https://github.com/maticnetwork/maticjs-plasma/blob/main/examples/erc20/transfer_matic.js) code given in example dir of this repository to transfer Native Matic token from one account to other account. Whenever I run the code the transaction is successful but the Matic is not transferred, and this is what I see on polygonscan

transaction hash - [0xcacf05fbcc7b32817e4ddc5c6cf6f19e9fdbb13a56826e6784b73a1f1e4e7bc5](https://mumbai.polygonscan.com/tx/0x8e82f0c84bfdc839d381dd53d0becb71bddc56c42474464bf9cf9c9050b692c5)

![image](https://user-images.githubusercontent.com/20986069/151284611-614049d0-178c-4a2c-923b-7d2e8a11b3c6.png)

I'm doing this on 

network - **_testnet_**
version - **_mumbai_**
From Private key - 
From address - https://mumbai.polygonscan.com/address/0x89A7D20d87146c36dfD72cB0B85C728E0C57E778
To address - https://mumbai.polygonscan.com/address/0xde55899b7f9091ef0db6ffd38ddd976493e17092
Root RPC - **_Goreli RPC_**
Child RPC - **_MATIC mumbai RPC_**

Have tried `multiple RPC and wallet key` combination and the RPC and wallet are on testnet (mumbai). I've also tried with `multiple Gas Limit` starting from the highest which gives error for high txn fees than cap, till low, getting same error for all the combinations.

I've enough **_MATIC_** in my wallet. (Do i also need ETH in my wallet ?)

_Important Information which rules out config issue_ - 

When i try transfer.js with `0x2d7882beDcbfDDce29Ba99965dd3cdF7fcB10A1e` TST token (ERC 20 token on Polygon, the transaction is successful and the token is getting transferred correctly from one wallet to other). So clearly looks like issue with transferring MATIC on testnet

@ujjwalgupta94 can you help here. Have been scratching my head since 2 days on this issue.
Let me know if you need any other information which will help in resolving the issue.

**Comment by ujjwalgupta94** [comment]
Your transaction is in pending state, it might be due to low gas or wrong nonce. Try sending the tx with higher gas fee.

**Comment by savannahar68** [comment]
@ujjwalgupta94 is there a way to estimate gas price using maticjs api ?

Ethers.js has a way to do it which resolved the issue - https://docs.ethers.io/v5/api/providers/provider/#Provider-estimateGas
If there's a way to do it using Matic.js then let me know, I can try it and update the code in example folder

**Comment by ujjwalgupta94** [comment]
@savannahar68 `estimateGas` calculates amount of the gas required not the gas price right ? 

from the ether docs - 
```
Returns an estimate of the amount of gas that would be required to submit transaction to the network.
```

**Comment by savannahar68** [comment]
@ujjwalgupta94 that's right
Do we have similar function in maticjs library ?

**Comment by ujjwalgupta94** [comment]
yes that is already calculated for each transaction, you can see for yourself by using `returnTransaction:true` in the transaction send option object.

like this - 

```
const result = await erc20Token.transfer(amount, to, { returnTransaction:true });
console.log('result',result);
```

this is how the result will look like - 

```
 {
  from: '0xbBca830ee5dCABde33DB24496b5524b9C5A69fe6',
  returnTransaction: true,
  gasLimit: 112590,
  nonce: 84,
  chainId: 5,
  data: '0xe3dec8fb000000000000000000000000bbca830ee5dcabde33db24496b5524b9c5a69fe60000000000000000000000002e3ef7931f2d0e4a7da3dea950ff3f19269d906300000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000007b0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000',
  to: '0xBbD7cBFA79faee899Eaf900F13C9065bF03B1A74'
}
```

The `gasLimit` is the one.
